tracer/src/Datadog.Trace/HttpOverStreams/HttpMessage.cs

```diff
@@ -45,20 +46,30 @@ public Encoding GetContentEncoding()
             }
 
             // text/plain; charset=utf-8
-            string[] pairs = contentType.Split(';');
-
-            foreach (string pair in pairs)
+            foreach (var pair in contentType.SplitIntoSpans(';'))
             {
-                string[] parts = pair.Split('=');
+                var parts = pair.AsSpan();
+                var index = parts.IndexOf('=');
 
-                if (parts.Length == 2 && string.Equals(parts[0].Trim(), "charset", System.StringComparison.OrdinalIgnoreCase))
+                if (index != -1)
                 {
-                    switch (parts[1].Trim())
+                    var firstPart = parts.Slice(0, index).Trim();
+
+                    if (!firstPart.Equals("charset".AsSpan(), StringComparison.OrdinalIgnoreCase))
+                    {
+                        continue;
+                    }
+
+                    var secondPart = parts.Slice(index + 1).Trim();
+
+                    if (secondPart.Equals("utf-8".AsSpan(), StringComparison.OrdinalIgnoreCase))
+                    {
+                        return Utf8Encoding;
```

Comment:
nit: we have `UTF8Encoding(encoderShouldEmitUTF8Identifier: false)` elsewhere, e.g. in `EncodingHelpers` (or something) we should probably change the code to use that?

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/5301/commits/f1af3493621372ca76eb2671624d5a782ad679df